packages/core/src/http_clients/got-scraping-http-client.ts

```diff
@@ -20,6 +20,31 @@ export class GotScrapingHttpClient implements BaseHttpClient {
         return !['CONNECT', 'connect'].includes(request.method!);
     }
 
+    private parseHeaders(headers: Record<string, string | string[] | undefined>): Headers {
+        const filterPseudoheaders = (headerPair: string[]): boolean => {
+            return !headerPair[0].startsWith(':');
+        };
+
+        const mergeMultipleHeaderValues = (header: [string, string | string[] | undefined]): string[][] => {
```

Comment:
If you rewrote this as a generator, you might not need to flatten the result and it would be more readable, too 🙂